### PR TITLE
Use compile specs for backend configuration

### PR DESCRIPTION
**Pull request by per**
Move the compiler flags passed to the backend up to the user in order to target different configurations.

### CODE REVIEW

**Comment by mergennachin** [comment]
@cccclai - could you take a look at this PR? 


@per did you run lintrunner by any chance? 

https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#coding-style

**Comment by per** [comment]
@mergennachin Yes, I did. All checks green.

File: backends/arm/arm_partitioner.py [hunk 1]
```diff
@@ -47,9 +53,10 @@ def is_node_supported(self, submodules, node: torch.fx.Node) -> bool:
 
 @final
 class ArmPartitioner(Partitioner):
-    compile_spec = []
+    compile_spec: List[CompileSpec]
 
-    def __init__(self) -> None:
+    def __init__(self, compile_spec: List[CompileSpec]) -> None:
+        self.compile_spec = compile_spec
         self.delegation_spec = DelegationSpec(ArmBackend.__name__, self.compile_spec)
```
Comment (cccclai) [inline backends/arm/arm_partitioner.py:60]: Looks like `self.compile_spec` is no longer needed?
```suggestion
        self.delegation_spec = DelegationSpec(ArmBackend.__name__, compile_spec)
```

Comment (cccclai) [reply backends/arm/arm_partitioner.py:60]: The rest part looks good to me

Comment (per) [reply backends/arm/arm_partitioner.py]: > Looks like `self.compile_spec` is no longer needed?

Done!

**Comment by cccclai** [comment]
new changes look good - thank you!

**Pull request closed without merging**